easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -247,8 +257,18 @@ private static void runAnnotatedMethods(Object object, int requestCode) {
 
     private static void checkCallingObjectSuitability(Object object) {
         // Make sure Object is an Activity or Fragment
-        if (!((object instanceof Fragment) || (object instanceof Activity))) {
-            throw new IllegalArgumentException("Caller must be an Activity or a Fragment.");
+        boolean isActivity = object instanceof Activity;
+        boolean isSupportFragment = object instanceof Fragment;
+        boolean isAppFragment = object instanceof android.app.Fragment;
+
+        if (!(isSupportFragment || isActivity ||
+                (isAppFragment && android.os.Build.VERSION.SDK_INT >= 23))) {
+            if (isAppFragment) {
+                throw new IllegalArgumentException(
```

Comment:
Now the library will crash the app if the developer tries to execute a permissions request but is using app fragments on API <  23.  This is good behavior, but we should protect the developer from getting here.

Could you add a check to `hasPermissions` that looks something like this:
```
if (Build.VERSION_SDK_INT >= Build.VERSION_CODES.M) {
    Log.w(TAG, "hasPermissions: API version < M, returning true by default");
    return true;
}
```

And a similar thing to `shouldShowRequestPermissionRationale` but returning `false` if the API is too low.

This way developers on lower API levels will be less likely to crash their app.  If they don't have the permissions they require in the Manifest, the system will crash them and they can deal with the error at that point.